src/librustc_hir/intravisit.rs

```diff
@@ -151,20 +151,20 @@ pub enum NestedVisitorMap<'this, M> {
     /// to use `visit_all_item_likes()` as an outer loop,
     /// and to have the visitor that visits the contents of each item
     /// using this setting.
-    OnlyBodies(&'this M),
+    OnlyBodies(M),
```

Comment:
I would prefer if we kept the reference here so that `M: ?Sized` was valid, specifically such that `dyn Map<'v>` was valid. (See https://github.com/rust-lang/rust/pull/69920.)

Reply:
I think you should be able to use `type Map = &'a dyn Map<'v>`.

Reply:
That would work I think.